dot/sync/chain_sync.go

```diff
@@ -37,6 +37,13 @@ const (
 	tip
 )
 
+type blockOrigin byte
+
+const (
+	networkInitialSync blockOrigin = iota
+	networkBroadcast
+)
```

Comment:
There are few more block origins in substrate https://github.com/paritytech/substrate/blob/86c6bb9614c437b63f3dbd2afddef52f32af7866/primitives/consensus/common/src/lib.rs#L62-L63.

I will leave it upto you 
- if you want to add them as part of this PR or 
- just keep the origins that you need and keep the issue open to do the rest later

Reply:
Yeah, I will keep only `NetworkInitialSync` and `NetworkBroadcast` since they are the ones in use right now (the sync package only deals with blocks of those kinds), we can introduce `ConsensusBroadcast`, `Own`, `File`, and `Genesis` when needed.